### PR TITLE
Endpoints to retrieve realtime filter actions

### PR DESCRIPTION
**Pull request by rbatistadev**
#### Type Of Change 
- [ ] Small bug fix
- [X] New feature or enhancement
- [ ] Breaking change

#### Checklist:
- [X] Commits are named and have tag following [commit rules](https://github.com/irontec/ivozprovider/blob/bleeding/doc/dev/en/commits.md)
- [X] Commits are split per component (schema, web/admin, kamusers, agis, ..)
- [X] Changes have been tested locally
- [ ] Fixes an existing issue (Fixes #XXXX) 
- [ ] Upport from existing Pull request #XXXX

#### Description
Created four new endpoints to retrieve filter actions for future use in Redis

### CODE REVIEW

File: web/rest/brand/config/api/resources.yml [hunk 1]
```diff
@@ -50,6 +50,32 @@ Model\ActiveCalls:
...
+        produces:
+        - 'application/json'
+        tags:
+        - My
+        parameters:
+        - name: b
+          in: query
+          type: string
+          required: false
+          description: 'Brand Id'
```
Comment (mmadariaga) [inline web/rest/brand/config/api/resources.yml:68]: brand API souldn't accept brand id

File: web/rest/brand/src/Controller/My/ActiveCallsRealtimeFilterAction.php [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+    public function __invoke(Request $request): Response
+    {
+        $token = $this->tokenStorage->getToken();
+
+        if (!$token || !$token->getUser()) {
+            throw new ResourceClassNotFoundException('User not found');
+        }
+
+        $brandId = $request->query->get('b');
```
Comment (mmadariaga) [inline web/rest/brand/src/Controller/My/ActiveCallsRealtimeFilterAction.php:28]: get from user

File: web/rest/brand/src/Controller/My/ActiveCallsRealtimeFilterAction.php [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+            $criteria['trunks']['c'] = $companyId;
+        }
+
+        if ($ddiProviderId) {
+            $criteria['trunks']['dp'] = $ddiProviderId;
+        }
+
+        if ($carrierId) {
+            $criteria['trunks']['cr'] = $carrierId;
+        }
```
Comment (mmadariaga) [inline web/rest/brand/src/Controller/My/ActiveCallsRealtimeFilterAction.php:58]: Extract into application service

File: web/rest/client/config/api/resources.yml [hunk 4]
```diff
@@ -102,6 +102,32 @@ Model\ActiveCalls:
...
+          - name: c
+            in: query
+            type: string
+            required: false
+            description: 'Company Id'
+          - name: cr
+            in: query
+            type: string
+            required: false
+            description: 'Carrier Id'
```
Comment (mmadariaga) [inline web/rest/client/config/api/resources.yml:130]: wrong filters for client API

File: web/rest/client/src/Controller/My/ActiveCallsRealtimeFilterAction.php [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+            'username' => $user->getUsername(),
+            'roles' => $user->getRoles()
+        ];
+
+        $criteria = $this->registrationChannelResolver->getDefaultCriteria($tokenPayload);
+        $result = [
+            'criteria' => $this
+                ->registrationChannelResolver
+                ->criteriaToString($tokenPayload, $criteria)
+        ];
```
Comment (mmadariaga) [inline web/rest/client/src/Controller/My/ActiveCallsRealtimeFilterAction.php:41]: extract into app service

File: web/rest/platform/config/api/resources.yml [hunk 6]
```diff
@@ -21,6 +21,32 @@ Model\ActiveCalls:
...
+      method: 'GET'
+      path: '/my/active_calls/realtime_filter'
+      route_name: 'get_my_active_calls_realtime_filter'
+      swagger_context:
+        produces:
+          - 'application/json'
+        tags:
+          - My
+        parameters:
+          - name: b
```
Comment (mmadariaga) [inline web/rest/platform/config/api/resources.yml:35]: some filters are missing

File: web/rest/platform/src/Controller/My/ActiveCallsRealtimeFilterAction.php [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+
+        if ($carrierId) {
+            $criteria['trunks']['cr'] = $carrierId;
+        }
+
+        $result = [
+            'criteria' => $this
+                ->registrationChannelResolver
+                ->criteriaToString($tokenPayload, $criteria)
+        ];
```
Comment (mmadariaga) [inline web/rest/platform/src/Controller/My/ActiveCallsRealtimeFilterAction.php:57]: extract

**Review by mmadariaga — CHANGES_REQUESTED** [review]

File: library/Ivoz/Provider/Application/Service/ActiveCall/RegistrationChannelResolver.php [hunk 8]
```diff
@@ -0,0 +1,345 @@
...
+                $defaultCriteria
+            )
+        ];
+    }
+
+    /**
+     * @param array{username: string, roles: array<int, string>} $tokenPayload
+     * @return array<'trunks'|'users', array<'b'|'c', int>>
+     */
+    public function getDefaultCriteria(array $tokenPayload): array
```
Comment (mmadariaga) [inline library/Ivoz/Provider/Application/Service/ActiveCall/RegistrationChannelResolver.php:50]: Services can only have one public method

File: web/rest/platform/src/Controller/My/ActiveCallsRealtimeFilterAction.php [hunk 9]
```diff
@@ -0,0 +1,65 @@
...
+
+        $defaultCriteria = $this
+            ->registrationChannelResolver
+            ->getDefaultCriteria($tokenPayload);
+
+        $filters = [
+            'b' => (int)$brandId,
+            'c' => (int)$companyId,
+            'cr' => (int)$carrierId,
+            'dp' => (int)$ddiProviderId
```
Comment (mmadariaga) [inline web/rest/platform/src/Controller/My/ActiveCallsRealtimeFilterAction.php:50]:
```suggestion
            'b' => (int) $brandId,
            'c' => (int) $companyId,
            'cr' => (int) $carrierId,
            'dp' => (int) $ddiProviderId
```

**Review by mmadariaga — CHANGES_REQUESTED** [review]

File: web/rest/platform/config/api/resources.yml [hunk 10]
```diff
@@ -21,6 +21,37 @@ Model\ActiveCalls:
...
+          - name: cr
+            in: query
+            type: string
+            required: false
+            description: 'Carrier Id'
+          - name: dp
+            in: query
+            type: string
+            required: false
+            description: 'DdiProvider Id'
```
[the remark this answers is not in the document]
Comment (rbatistadev) [reply web/rest/platform/config/api/resources.yml:54]: Set inbound and outbound as new query arguments for direction

**Comment by mmadariaga** [comment]
Add API tests and you're done

**Review by mmadariaga — APPROVED** [review]

**Pull request merged**